src/controllers/mailing-list.ts

```diff
@@ -0,0 +1,86 @@
+import express, {NextFunction, Request, Response} from 'express';
+import SheetsService from "../lib/sheets/SheetsService";
+import * as EmailValidator from 'email-validator';
+import fs from 'fs';
+
+export const mailingList = async function(req: Request, res: Response) {
+  const personalName = req.body.personalName;
+  const organisationName = req.body.organisationName;
+  const contactEmail = req.body.contactEmail;
+  const serviceName = req.body.serviceName;
+  let errorMessages: Map<String, String> = new Map<String, String>();
+
+  const formValueHolder = {
+    personalNameHolder: personalName,
+    organisationNameHolder: organisationName,
+    contactEmailHolder: contactEmail,
+    serviceNamehHolder: serviceName
+  }
+
+  const onlyLettersPattern = /^[a-zA-Z\-\s]{1,300}$/;
+  const onlyLettersAndNumbersPattern = /^[a-zA-Z\-0-9_\s]{1,300}$/;
+
+  if(!onlyLettersPattern.test(personalName)) {
+    errorMessages.set('personalName', 'Enter your name');
+  }
+
+  if(!onlyLettersAndNumbersPattern.test(organisationName)) {
+    errorMessages.set('organisationName', 'Enter your organisation name');
+  }
+
+  if(contactEmail === "") {
+    errorMessages.set('contactEmail', 'Enter your government email address');
+  } else if(!EmailValidator.validate(contactEmail)){
+    errorMessages.set('contactEmail', 'Enter an email address in the correct format, like name@example.gov.uk');
+  } else {
+    try {
+      const emails = fs.readFileSync("./valid-email-domains.txt", "utf-8");
```

Comment:
This will need to be `async` as it's in the middle of a controller.  Ideally we'll only load the file once at application start up.  Where we use it elsewhere and it's synchronous we can get away with it because it happens once when the application is starting but will eventually be refactored to be asynchronous anyway.  As we will be refactoring, the file is relatively small and we don't expect high volumes of traffic this can stay where it is as long as we make it an `async / await` operation.

`import {promises as fs} from 'fs';` should help